packages/react-client/src/ReactFlightClient.js

```diff
@@ -331,6 +339,12 @@ export function parseModelTuple(
   const tuple: [mixed, mixed, mixed, mixed] = (value: any);
 
   if (tuple[0] === REACT_ELEMENT_TYPE) {
+    if ((tuple[1]: any).$$typeof === REACT_PROVIDER_TYPE) {
+      return createElement(ServerContextWrapper, null, {
+        ServerContext: ((tuple[1]: any): ReactProviderType<any>)._context,
+        ...tuple[3],
```

Comment:
You know it's going to be children and value, so you can just enumerate those here. It's faster than spread.